
### Examining racial discrimination in the US job market

#### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

#### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes.

#### Exercise
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Discuss statistical significance.

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

****

In [49]:
import pandas as pd
import numpy as np
from scipy import stats

useFullPath=True;
if useFullPath:
    data = pd.io.stata.read_stata('C:\Users\dwhitney\Desktop\Springboard Course\Course Notes\Mini-projects\statistics project 2\data\us_job_market_discrimination.dta')    
else:
    data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

# 1.) What test is appropriate for this problem? Does CLT apply?
A z-test is the most appropriate choice for this problem. This is because the central limit theorem does apply to this problem because the theorem states that the sum of a large number of independent, identically distributed variables will be approximately normal, **regardless of the underlying distribution** (even binomial distributions).

# 2.) What are the null and alternate hypotheses?
The null hypothesis would state that race is not a significant factor in the number of callbacks, whereas the alternate hypothesis would claim that race is a significant factor (two-tail). To test between these two possibilities, we can construct a null distribution where the percentage of callblacks is blind to race. We find that the base probability of a callbacks (independent of race) is:

In [56]:
p = sum(data.call)/len(data);
"Percentage of callbacks = {:5.4}%".format(100*p)

'Percentage of callbacks = 8.049%'

# 3.) Compute margin of error, confidence interval, and p-value.
Now we can test whether the number of race-related callbacks (our alternatate hypothesis) is different from our null distribution. We will specifically use a one-sample z-test to test whether the callback for blacks significantly differs from a null distribution, in which callbacks are blind to race. The parameters for the null distribution will be that: a.) the distribution is normal, b.) the number of sample will be matched to the number of black related calls (n=2435), and c.) the probability of a successful callback is blind to race (p=8.049%).

In [57]:
from IPython.display import display

n_CallBacks = sum(data[data.race=='b'].call);
n = sum(data.race=='b');
p = sum(data.call)/len(data);
meanOfNormalDistribution = n*p;
stdDeviationOfNormalDistribution = np.sqrt(n*p*(1-p));
zStatistic = np.abs((n_CallBacks - meanOfNormalDistribution)/stdDeviationOfNormalDistribution)
display("zStatistic = {}".format(zStatistic))

'zStatistic = 2.9050860929'

As the computed z-statistic is greater than 1.96, the critical value of significance for our stastistical test (95% confidence, where a=0.1), we can then conclude that race significantly contributes the rate of callbacks. Next we'll compute how the number of race-related callbacks compares against the margin of error and 95% confidence intervals for our null distribution:

In [58]:
marginOfError=1.96*(stdDeviationOfNormalDistribution/np.sqrt(n));
display('Actual number of race-related callbacks = {}'.format(n_CallBacks))
display('Margin of error for null distribution = {:04.2f}'.format(marginOfError));
display('95% confidence intervals for null distribution = [{:04.2f},{:04.2f}]'.format(meanOfNormalDistribution-marginOfError,meanOfNormalDistribution+marginOfError))

'Actual number of race-related callbacks = 157.0'

'Margin of error for null distribution = 0.53'

'95% confidence intervals for null distribution = [195.47,196.53]'

Next we'll compute how significant our effect is by converting our z-statistic to a p-value:

In [59]:
p_value = stats.norm.sf(abs(zStatistic))*2 #twosided
display('p-value = {}'.format(p_value))

'p-value = 0.00367152095404'

# 4.) Discuss statistical significance. 
As our p-value is 0.003, the chance of this effect arising from random fluctuations is less than 1 in 272. Per convention statistically significant effects must have p-values less than either p=0.01 or p=0.05. As our p-value is much lower than either of these cutoffs, we can safely conclude that the number of race-related callbacks for blacks is significantly less than would be expected if race was not a contributing factor.